src/Elasticsearch.Net/Configuration/ConnectionConfiguration.cs

```diff
@@ -54,8 +47,9 @@ public ConnectionConfiguration(IConnectionPool connectionPool, IConnection conne
 	public abstract class ConnectionConfigurationBase<TConnectionConfiguration> : TransportConfigurationBase<TConnectionConfiguration>, IConnectionConfigurationValues
 		where TConnectionConfiguration : ConnectionConfigurationBase<TConnectionConfiguration>, IConnectionConfigurationValues
 	{
-		protected ConnectionConfigurationBase(IConnectionPool connectionPool, IConnection connection, ITransportSerializer serializer)
-			: base(connectionPool, connection, serializer, ElasticsearchProductRegistration.Default) =>
+		protected ConnectionConfigurationBase(IConnectionPool connectionPool, IConnection connection, ITransportSerializer serializer,
+			IProductRegistration registration = null)
```

Comment:
Is there a case we imagine where someone would want to pass in a different `IProductRegistration` within the client context?

Reply:
aye `NEST` uses a slightly different registration then the low level client :)